[clang][AArch64][FMV] Stop emitting alias to ifunc.

labrinea opened the pull request:
Long story short the interaction of two optimizations happening in GlobalOpt results in a crash. For more details look at the issue https://github.com/llvm/llvm-project/issues/96197. I will be fixing this in GlobalOpt but it is a conservative solution since it won't allow us to optimize resolvers which return a pointer to a function whose definition is in another TU when compiling without LTO:

```
__attribute__((target_version("simd"))) void bar(void);
__attribute__((target_version("default"))) void bar(void);
int foo() { bar(); }
```

fixes: #96197

labrinea committed:
[clang][AArch64][FMV] Stop emitting alias to ifunc.

Long story short the interaction of two optimizations happening in
GlobalOpt results in a crash. For more details look at the issue
https://github.com/llvm/llvm-project/issues/96197. I will be
fixing this in GlobalOpt but it is a conservative solution since
it won't allow us to optimize resolvers which return a pointer
to a function whose definition is in another TU when compiling
without LTO:

__attribute__((target_version("simd"))) void bar(void);
__attribute__((target_version("default"))) void bar(void);
int foo() { bar(); }

jroelofs commented:
cc @tahonermann

jroelofs commented:
I think this needs a release note.

DanielKristofKiss reviewed (approved):
Second the relnote otherwise LGTM.

labrinea committed:
[clang[AArch64][FMV] Release notes for ABI breaking removal of alias.

We removed the global alias that was pointing to AArch64 Function
Multiversioning ifuncs because it interacts badly with GlobalOpt.

labrinea committed:
Merge branch 'main' into remove-ifunc-alias

labrinea committed:
Merge branch 'main' into remove-ifunc-alias

labrinea committed:
Merge branch 'main' into remove-ifunc-alias
